chore: Replace widget skeleton with internal swap skeleton

tinaszheng opened the pull request:
## Description
Removes the dependency for rendering a Swap skeleton imported from @uniswap/widgets 

Received some new designs from @infredible - https://www.figma.com/file/kNSDBMpOzxSTOP6MerohLm/Web-Design-Spec?type=design&node-id=13335-127826&t=CAu5FD8kCowy3z7c-0 

I haven't finished the whole thing ^ but close enough equivalent for now. Ticket for finishing it: WEB-2030

<!-- Delete this section if your change does not affect UI. -->

| Before       | After (Desktop) | After (Mobile) |
| ------------ |---------------- | -------------- |
|<img width="400" alt="image" src="https://github.com/Uniswap/interface/assets/59578595/3fc3b3d0-9ecc-4743-b6f0-dc6a5124f300">|<img width="406" alt="image" src="https://github.com/Uniswap/interface/assets/59578595/b67fed19-8b2e-4b2b-8a0d-8ebbde220e48">| not shown on mobile  |

Note I updated the Swap arrow to be the same one as in the Swap component


### QA (ie manual testing)
If you want to see this on Vercel, I recommend going to the token details page, then going to Network > Throttling > No network > REFRESH to see the loading state show

### Automated testing
Added snapshot test

tinaszheng commented on src/components/swap/SwapSkeleton.tsx:
```diff
@@ -0,0 +1,103 @@
+  const theme = useTheme()
+
+  return (
+    <LoadingWrapper>
```
note the general structure was copied over from https://github.com/Uniswap/widgets/blob/main/src/components/Swap/Skeleton.tsx - but made some simplifications

tinaszheng commented:
@zzmp would you mind taking a look at this failing cypress test? it seems vaguely related to mike's test failure from earlier today, but im unsure

cmcewen commented on src/components/swap/SwapSkeleton.tsx:
```diff
@@ -0,0 +1,103 @@
+
+const Blob = styled.div<{ height: string; width: string; radius?: number; marginTop?: number }>`
+  background-color: ${({ theme }) => theme.backgroundModule};
+  border-radius: ${({ radius }) => (radius ?? 0.25) + 'rem'};
```
it seems like we settled on `rem` - is there anywhere in readme/notion we can point people to as the "why" and some migration tips

grabbou replied:
I still use pixels in my PR 😆

cmcewen commented on src/components/swap/SwapSkeleton.tsx:
```diff
@@ -0,0 +1,103 @@
+}
+
+function FloatingButton() {
+  return <Blob marginTop={0.55} height="3.5rem" width="100%" radius={0.75} />
```
this component api is a little confusing to me - it's not clear that `marginTop` is a number in rem, and it's inconsistent with the type of height/width which are strings. i'd vote either making everything a string or every prop a number with a `Rem` at the end

grabbou replied:
To be honest, this is perfect example where `rem` is confusing. I have no idea what `3.5rem` is here.

cmcewen reviewed (approved):
it's not a blocker for me but `Blob` props could be more consistent

grabbou commented on cypress.config.ts:
```diff
@@ -5,7 +5,6 @@ import { setupHardhatEvents } from 'cypress-hardhat'
 export default defineConfig({
   projectId: 'yp82ef',
   videoUploadOnPasses: false,
-  waitForAnimations: false,
```
Is this intentional? I think it was added recently by Zach to speed things up

tinaszheng replied:
yeah it was to fix a flakey test but this has been added on main already so I can rebase

grabbou replied:
Ah, sorry, I misread. I thought it was the opposite. Yeah, this is good change!

grabbou commented on src/pages/Swap/index.tsx:
```diff
@@ -65,7 +65,8 @@ import { computeFiatValuePriceImpact } from '../../utils/computeFiatValuePriceIm
 import { supportedChainId } from '../../utils/supportedChainId'
-const ArrowContainer = styled.div`
+
+export const ArrowContainer = styled.div`
```
nit: can we move this to a separate file? Ideally, we could have the `ArrowDown` component exported from `theme/components` and the fact that it uses `ArrowContainer` would just be an implementation detail. It's quite confusing to import just the container to wrap your icon.

Update: Maybe `styleds.tsx` can work too.

tinaszheng replied:
can i do in a followup :D since this is already existing convention in the main Swap flow.

grabbou replied:
Yes, please! Just make sure to not forget (I'll check hahaha jk)

grabbou commented on src/components/swap/SwapSkeleton.tsx:
```diff
@@ -0,0 +1,103 @@
+        <FloatingInput />
+      </InputColumn>
+      <div>
+        <ArrowWrapper clickable={false}>
```
nit: I would prefer if this was just an `Arrow` component, the fact we need wrapper/container/arrowdown is super confusing to me.

tinaszheng replied:
can i refactor this in a separate PR :D beacuse it's the same in the main Swap component and i don't want to make any changes there for this PR yet.

grabbou commented on src/pages/Swap/index.tsx:
```diff
@@ -65,7 +65,8 @@ import { computeFiatValuePriceImpact } from '../../utils/computeFiatValuePriceIm
-const ArrowContainer = styled.div`
+
+export const ArrowContainer = styled.div`
   display: inline-block;
```
Why does this component have display defined twice? I now it's not touched in this diff, but would be cool to clean up. Might be worth checking whether width and height 100% are required to be present as well, b/c that looks like a left-over.

tinaszheng replied:
omg 😂

grabbou commented on src/components/Widget/index.tsx:
```diff
@@ -198,7 +198,7 @@ export default function Widget({
-export function WidgetSkeleton({ width = DEFAULT_WIDGET_WIDTH }: { width?: number | string }) {
+function WidgetSkeleton({ width = DEFAULT_WIDGET_WIDTH }: { width?: number | string }) {
   const theme = useWidgetTheme()
   return <SwapWidgetSkeleton theme={theme} width={width} />
```
Here, I think we're still rendering skeleton from Widget. Is this intentional?

grabbou replied:
Sounds good!

grabbou commented on src/components/swap/SwapSkeleton.tsx:
```diff
@@ -0,0 +1,103 @@
+      </InputColumn>
+      <div>
+        <ArrowWrapper clickable={false}>
+          <ArrowContainer color={theme.textPrimary}>
```
nit: do we really need `color` for `ArrowContainer`?

grabbou commented on src/components/swap/SwapSkeleton.tsx:
```diff
@@ -0,0 +1,103 @@
+      <div>
+        <ArrowWrapper clickable={false}>
+          <ArrowContainer color={theme.textPrimary}>
+            <ArrowDown size="16" color={theme.textTertiary} />
```
nit: I would recommend wrapping this with a StyledComponent and passing `color` property there. I know it's kind of a nit pick, but if you remember my previous PR, I did the same for the Gas Icon, so it just keeps it more consistent.

grabbou commented on src/components/Widget/index.tsx:
```diff
@@ -36,7 +36,7 @@ import { useSyncWidgetSettings } from './settings'
 import { useSyncWidgetTransactions } from './transactions'
 
-export const DEFAULT_WIDGET_WIDTH = 360
+const DEFAULT_WIDGET_WIDTH = 360
```
nit: it would be great to keep this in sync with `TokenDetails/Skeleton`, so they never run out of sync. 

In reality, this is just bad design, because it seems to me that we have two different components, TokenDetails "main page" and TokenDetails "skeleton".

I believe, instead, we should have a single component, TokenDetails, where we would have "RightPanel" (with fixed width) and inside, based on the loading state, we would either render WidgetSkeleton or Widget. That way, both WidgetSkeleton and Widget would not have to set width, but use 100% of available space and delegate it back to RightPanel to decide. 

I don't know whether this change is possible within this PR, probably not, but I would leave a TODO comment + Jira ticket to refactor this, because I think this can be improved.

tinaszheng replied:
agreed i was also really confused by it when i was refactoring! will make a linear ticket

grabbou replied:
I usually add it to the code too, but don't want you to re-request another review from Connor so just for the future haha

grabbou reviewed (commented):
Looks good! Left some comments, mostly nits, I think that we could do some further tweaking here and there (if time allows), otherwise, it's fine!

tinaszheng commented:
@cmcewen @grabbou made a few changes, please take a lewwwk~~~~

mainly i received some updated designs from Fred, so i incorporated some of that into this PR. 
- moved to using px because we actually don't have a strongly defined reason for moving to rem right now, so sticking with px until we come up with a migration strategy. as it is, the mixture of rem + px doesn't quite make sense to me, so i'm siding with @grabbou to stick with px for now
- refactored the components a bit to take advantage of flex gap instead of using one-off padding values to do layout. i feel like the new ver is much cleaner~
- removed the need for some props

cmcewen reviewed (approved):
🚢
